event-log/README.md

```diff
@@ -0,0 +1,243 @@
+# event-log
+
+This is a microservice which provides CRUD operations for Event Log DB.
+
+## API
+
+| Method | Path                                     | Description                                                    |
+|--------|------------------------------------------|----------------------------------------------------------------|
+|  POST  | ```/events```                            | Creates an event with a `NEW` status                           |
+|  PATCH | ```/events```                            | Changes events' data by applying the given patch               |
+|  GET   | ```/events/latest```                     | Finds events for all the projects with the latest `event_date` |
+|  PATCH | ```/events/:id/projects/:id/status```    | Updates event status                                           |
+|  GET   | ```/events/projects/:id/status```        | Finds processing status of events belonging to a project       |
```

Comment:
This could be `GET /events?project_id=<project-id>`, parsing the response to get the processing stati should in this case be the responsibility of the client in my opinion.

Reply:
Your suggestion made me think that there's a bigger problem with that resource. The naming suggest that it simply returns some project events' data but in fact, it gives very specific info about the processing status of events belonging to a project. Wouldn't that then be better if we go with smth like this:
`GET /events/processing-status?project_id=<project-id>`
?

Reply:
that doesn't seem too restful unless you have `/events/:id/processing-status` and you get the `:id` from `/events?project_id=<project-id>`

Reply:
So would maybe `/events/projects/:id/processing-status` work the best? The way I see it is that we don't filter the results here but simply return some specific data for some events of a certain project.

Reply:
but there is also `/events/:id/projects/:id/status` - I think this mixing makes it confusing about which belongs to what

Reply:
Agreed. What about `/events/projects/:id/processing-progress`?

Reply:
I think this gets us back to @ableuler's initial question - why not just put the `processing-progress` into the body? then the endpoint can be just `/events/projects/:id`

Reply:
Well, a `GET` with a body does not seem that right. Here's what people say in StackOverflow citing HTTP specs:

```
The HTTP specification says in section 4.3

A message-body MUST NOT be included in a request if the specification of the request method (section 5.1.1) does not allow sending an entity-body in requests.

Section 5.1.1 redirects us to section 9.x for the various methods. None of them explicitly prohibit the inclusion of a message body. However...

Section 5.2 says

The exact resource identified by an Internet request is determined by examining both the Request-URI and the Host header field.

and Section 9.3 says

The GET method means retrieve whatever information (in the form of an entity) is identified by the Request-URI.
```

Reply:
sorry I didn't mean in the request body, I meant that it would be returned in the list of events and their properties.

Reply:
I think @rokroskar meant the response and not the request body. For me there are two clean RESTful options: 
- `GET /events?project_id=<project-id>` which should return a list of events, the processing status would be one field of an event.
- `GET /statuses?project_id=<project-id>` which returns a list of statuses, statuses therefore become resources just like events.

Reply:
`/processing-status?projects-id=<id>`